packages/ui/src/components/va-data-table/hooks/useTableScroll.ts

```diff
@@ -0,0 +1,65 @@
+import { getCurrentInstance, computed, ExtractPropTypes } from 'vue'
+
+import { useIntersectionObserver, useElementRef } from '../../../composables'
+
+export const useTableScrollProps = {
+  scrollTopMargin: { type: Number, default: 0 },
+  scrollBottomMargin: { type: Number, default: 0 },
+}
+
+export const useTableScrollEmits = ['scroll:top', 'scroll:bottom']
+
+export default function useTableScroll (
+  props: ExtractPropTypes<typeof useTableScrollProps>,
+  emit: (event: 'scroll:bottom' | 'scroll:top', ...args: any[]) => void,
+) {
+  // TODO: replace 'vNodeProps' with 'attrs' when we bump vue to ^3.2
```

Comment:
Will it work with 3.2 now?

Reply:
The vue 3 [docs](https://v3-migration.vuejs.org/breaking-changes/listeners-removed.html) say listeners are available in `attrs`, but they are still not available in our current version of vue.
We have decided to use `getCurrentInstance()?.vnode.props` now and track listener availability in future vue versions.